drivers/atwinc15x0/atwinc15x0_netdev.c

```diff
@@ -646,7 +646,7 @@ static int _atwinc15x0_get(netdev_t *netdev, netopt_t opt, void *val,
     assert(dev == atwinc15x0);
 
     DEBUG("%s dev=%p opt=%u val=%p max_len=%u\n", __func__,
-          (void *)netdev, opt, val, max_len);
+          (void *)netdev, opt, val, (unsigned)max_len);
```

Comment:
is `sizeof(unsigned) >= sizeof(size_t)` --  i am not sure (ul seem like the least maybe sizeof(void*)-> PRI*PTR )
even thought one would assume any thing we actually got there should be fit for mcu there

Reply:
Sorry, I don't understand your comment. Can you rephrase?

Reply:
There is a max_len in size_t going in. With x86_64  this is of size 64Bit but this cast is to 32Bit - only some of the bits are used in that print / DEBUG -> this might hide the exact issue one is debugging.

PRI*PTR might be the right size print in most cases ( in some it might be to large) or a little less wordy lu or ul  which would might be fit for every thing we do (not sure).

lu is also the implementation the newlib defaults to for printing so no code size increase of any arch using that.

For u I know of one case it is to small to catch all of size_t.

The second part of the sentence is about the assumption of fzi-haxel that max_len is unlikely to exceed u on x86_64 because most of our systems are smaller and therefore any valid len is likely to fit into unsigned

I picked on this by on example there are multiple instances of this change some of them are more likely behave wrong than in this static function.